sstables/compaction.cc

```diff
@@ -1225,7 +1225,9 @@ class resharding_compaction final : public compaction {
     void backlog_tracker_adjust_charges() override { }
 
     shared_sstable create_new_sstable() const override {
-        return _sstable_creator(_shard);
+        // create_new_sstables is used only from the garbage_collected writer.
+        // It it not supposed to work with resharding compactions
+        throw std::runtime_error("create_new_sstable called from resharding. Shouldn't have happened!!");
```

Comment:
If this happens, the node goes into a death spiral, no?

Better to abort in this case.

@raphaelsc why would resharding not call create_new_sstable()?

Reply:
Raphael has already explained why not in a previous comment. This is already a reaction to his request.

The reason is that this function is only called by garbage collection sstables, which is only used for run-based compactions and never by resharding.

This function shouldn't exist. A cleanup should get rid of it.

I will change this to an abort.

Reply:
Please add the explanation to the changelog.

Reply:
This explanation was already in the changelog of this patch.
I can add it to the PR changelog.